### ISSUE TITLE
500 when server redeploys with different secret key

### ISSUE THREAD

**Issue by bkabrda** [puiterwijk/flask-oidc#55]
Hi,
because of a security incident, I had to replace `SECRET_KEY` of my server (Pagure 4.0.3) using `flask-oidc` 1.4.0. On redeploy with the new secret key, previously logged in users started getting this error:

```
[Thu May 24 09:00:44.419300 2018] [:error] [pid 31] 2018-05-24 09:00:44,351 [ERROR] flask.app: Exception on /rpms/setup [GET]
[Thu May 24 09:00:44.419316 2018] [:error] [pid 31] Traceback (most recent call last):
[Thu May 24 09:00:44.419318 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask/app.py", line 2292, in wsgi_app
[Thu May 24 09:00:44.419320 2018] [:error] [pid 31]     response = self.full_dispatch_request()
[Thu May 24 09:00:44.419322 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1815, in full_dispatch_request
[Thu May 24 09:00:44.419324 2018] [:error] [pid 31]     rv = self.handle_user_exception(e)
[Thu May 24 09:00:44.419326 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1718, in handle_user_exception
[Thu May 24 09:00:44.419328 2018] [:error] [pid 31]     reraise(exc_type, exc_value, tb)
[Thu May 24 09:00:44.419330 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask/app.py", line 1811, in full_dispatch_request
[Thu May 24 09:00:44.419332 2018] [:error] [pid 31]     rv = self.preprocess_request()
[Thu May 24 09:00:44.419333 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask/app.py", line 2087, in preprocess_request
[Thu May 24 09:00:44.419335 2018] [:error] [pid 31]     rv = func()
[Thu May 24 09:00:44.419337 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask_oidc/__init__.py", line 406, in _before_request
[Thu May 24 09:00:44.419338 2018] [:error] [pid 31]     self.authenticate_or_redirect()
[Thu May 24 09:00:44.419340 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask_oidc/__init__.py", line 426, in authenticate_or_redirect
[Thu May 24 09:00:44.419342 2018] [:error] [pid 31]     id_token = self._get_cookie_id_token()
[Thu May 24 09:00:44.419344 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/flask_oidc/__init__.py", line 352, in _get_cookie_id_token
[Thu May 24 09:00:44.419346 2018] [:error] [pid 31]     return self.cookie_serializer.loads(id_token_cookie)
[Thu May 24 09:00:44.419347 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/itsdangerous.py", line 795, in loads
[Thu May 24 09:00:44.419349 2018] [:error] [pid 31]     self, s, salt, return_header=True)
[Thu May 24 09:00:44.419351 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/itsdangerous.py", line 749, in loads
[Thu May 24 09:00:44.419353 2018] [:error] [pid 31]     self.make_signer(salt, self.algorithm).unsign(want_bytes(s)),
[Thu May 24 09:00:44.419354 2018] [:error] [pid 31]   File "/usr/lib/python2.7/site-packages/itsdangerous.py", line 374, in unsign
[Thu May 24 09:00:44.419356 2018] [:error] [pid 31]     payload=value)
[Thu May 24 09:00:44.419358 2018] [:error] [pid 31] BadSignature: Signature 'o3c13YeSK_1_Mqrf9t7HoCi-tSXlkCBl6fW2JC0RHA8' does not match
```

Users who weren't logged in previously could access the server and log in just fine. I think this error should be caught by flask-oidc and user logged out as consequence of it. Does that make sense?

**Comment by puiterwijk** [comment]
This got fixed in 0ee31ff32fd10398eca80e690c3194967e29fe6d.
Thanks again for the patch!

**Comment by wesparish** [comment]
@puiterwijk Can we get this tagged and uploaded to pypi? Thanks!
